Sources/ArcGISToolkit/Components/Offline/PreplannedMetadataView.swift

```diff
@@ -0,0 +1,118 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct PreplannedMetadataView: View {
+    /// The view model for the preplanned map.
+    @ObservedObject var model: PreplannedMapModel
+    
+    /// The action to dismiss the view.
+    @Environment(\.dismiss) private var dismiss
+    
+    var body: some View {
+        Form {
+            Section {
+                if let image = model.preplannedMapArea.thumbnail {
+                    HStack {
+                        Spacer()
+                        LoadableImageView(loadableImage: image)
+                            .clipShape(.rect(cornerRadius: 10))
+                            .padding(.vertical, 10)
+                        Spacer()
+                    }
+                }
+                VStack(alignment: .leading) {
+                    Text("Name")
+                        .font(.caption)
+                        .foregroundStyle(.secondary)
+                    Text(model.preplannedMapArea.title)
+                        .font(.subheadline)
+                }
+                VStack(alignment: .leading) {
+                    Text("Description")
+                        .font(.caption)
+                        .foregroundStyle(.secondary)
+                    if model.preplannedMapArea.description.isEmpty {
+                        Text("This area has no description.")
+                            .font(.subheadline)
+                            .foregroundStyle(.tertiary)
+                    } else {
+                        Text(model.preplannedMapArea.description)
+                            .font(.subheadline)
+                    }
+                }
+                VStack(alignment: .leading) {
+                    Text("Size")
+                        .font(.caption)
+                        .foregroundStyle(.secondary)
+                    Text(model.directorySize.formatted(.byteCount(style: .file, allowedUnits: [.kb, .mb])))
```

Comment:
Remember to left SwiftUI decide when a value needs to be re-formatted and only call a `formatted` method yourself when absolutely necessary.

```suggestion
                    Text(model.directorySize, format: .byteCount(style: .file, allowedUnits: [.kb, .mb]))
```

Reply:
For some reason, if I apply this change, it asks me to convert the `Int` to `Int64`, such as `Int64(model.directorySize), format: .byteCount(style: .file, allowedUnits: [.kb, .mb])` while the `formatted` method doesn't require so.

I'll convert the data type.

Reply:
Interesting. I wonder why the compiler didn't complain when I tried it? 🤔